### PR TITLE
chore: migrate from altool to notarytool

### PR DESCRIPTION
**Pull request by k9ert**
This PR updates the MACOS signing procedure from `xcrun altool`to `xcrun notarytool`.

### CODE REVIEW

File: utils/build-common.sh [hunk 1]
```diff
@@ -113,8 +120,18 @@ function building_electron_app {
 }
 
 function macos_code_sign {
+    # prerequisites for this:
+    # in short:
+    # * make sure you have a proper app-specific password on https://appleid.apple.com/account/manage
+    # * collect some information via scrun altool --list-providers -u "<yourAppleID>"
```
Comment (moneymanolis) [inline utils/build-common.sh:126]: `scrun` is definitely a type. Also, aren't transitioning away from `altool` in this PR? In any case, the command was not available for me.

**Comment by moneymanolis** [comment]
Some comments and TODOs

- [ ] Why are there two `create_virtualenv_for_pyinstaller` definitions (in build-common and build-osx)?
- [ ] Make `--apple-id `in `macos_code_sign` accept a shell argument (it is hard-coded right now)
- [ ]

File: utils/build-common.sh (hunk 1 above)
Comment (k9ert) [reply utils/build-common.sh:126]: True. However the information (TeamID) can also be found somewhere else.

**Comment by k9ert** [comment]
This PR is on hold until we have solved the issues with releasing on M2 machines.

**Comment by k9ert** [comment]
No longer needed as all changes were done in #2432

**Pull request closed without merging**